src/Product/AttributeMappingHelper.php

```diff
@@ -0,0 +1,89 @@
+<?php
+declare( strict_types=1 );
+
+namespace Automattic\WooCommerce\GoogleListingsAndAds\Product;
+
+use Automattic\WooCommerce\GoogleListingsAndAds\Infrastructure\Service;
+use Automattic\WooCommerce\GoogleListingsAndAds\Product\Attributes\Adult;
+use Automattic\WooCommerce\GoogleListingsAndAds\Product\Attributes\AgeGroup;
+use Automattic\WooCommerce\GoogleListingsAndAds\Product\Attributes\Color;
+
+defined( 'ABSPATH' ) || exit;
+
+/**
+ * Helper Class for Attribute Mapping
+ *
+ * @package Automattic\WooCommerce\GoogleListingsAndAds\Product
+ */
+class AttributeMappingHelper implements Service {
+
+
+	private const ATTRIBUTES_AVAILABLE_FOR_MAPPING = [
+		Adult::class,
+		AgeGroup::class,
+		Color::class,
+	];
+
+	/**
+	 * Gets all the available destinations
+	 *
+	 * @return array
+	 */
+	public function get_destinations(): array {
+		$destinations = [];
+
+		foreach ( self::ATTRIBUTES_AVAILABLE_FOR_MAPPING as $attribute ) {
+			$destinations[ $attribute::get_id() ] = $attribute::get_name();
+		}
+
+		return $destinations;
+	}
+
+	/**
+	 * Gets all the available sources identified by destination key
+	 *
+	 * @return array
+	 */
+	public function get_sources(): array {
+		$sources = [];
+
+		foreach ( self::ATTRIBUTES_AVAILABLE_FOR_MAPPING as $attribute ) {
+			$sources[ $attribute::get_id() ] = $attribute::get_sources();
+		}
+
+		return $sources;
+	}
+
+	/**
+	 * Gets the taxonomies and global attributes to render them as options in the frontend.
+	 *
+	 * @return array An array with the taxonomies and global attributes
+	 */
+	public static function get_source_taxonomies(): array {
+		$taxonomies = get_object_taxonomies( 'product' );
+		$taxes      = [];
+		$attributes = [];
+		foreach ( $taxonomies as $taxonomy ) {
+			$tax = get_taxonomy( $taxonomy );
+			if ( taxonomy_is_product_attribute( $taxonomy ) ) {
+				$attributes[ 'tax:' . $taxonomy ] = $tax->labels->name;
+				continue;
+			}
+
+			$taxes[ 'tax:' . $taxonomy ] = $tax->labels->name;
+		}
+		asort( $taxes );
+		asort( $attributes );
+
+		return array_merge(
```

Comment:
This list can potentially get really long. What kind of UI elements are we going to have for this? Does it include matching upon typing the word. Or would it be better to pick the most recommended attribute from the list.

For example I have a product attribute called `color` but I don't want to scroll through a large list of attributes to find it. If an attribute with matching name is found, should we pre-select it or bump it to the top of the list?

Reply:
Unfortunately as far as I know we don't receive yet from Google /.UX which attributes and sources we should have available.

This is actually how Google Feed manages this... so thats why I used this strategy.

>  What kind of UI elements are we going to have for this? Does it include matching upon typing the word.

Initially... Just a standard selector without search as I see in Figma... but we are still on i2 so. It could be a suggestion to add search in the selector. 

> Or would it be better to pick the most recommended attribute from the list.

This is a cool solution in my opinion to have a "Recommended" section in the selector. 

I suggest to elevate this to P2 discussion and modify this function in a future PR if needed.

Thoughts?

Reply:
Elevate to P2 and future PR sounds like a good plan.